File: TouchControl/src/io/github/mattson543/touchcontrol/display/ImagePanel.java
/**
 * Class Description:
 * Panel - holds image to display in GUI
 */

package io.github.mattson543.touchcontrol.display;

import java.awt.*;
import java.awt.image.BufferedImage;

import javax.swing.JPanel;

@SuppressWarnings("serial")
public class ImagePanel extends JPanel
{
	private BufferedImage image;

	//Constructors
	public ImagePanel()
	{
		super();
	}

	//Setters

	//Getters

	//Class methods
	public void setImage(BufferedImage image)
	{
		this.image = image;
	}

	@Override
	public void paintComponent(Graphics g)
	{
		super.paintComponent(g);

		if (image != null)
		{
			//Get image dimensions
			int width = image.getWidth();
			int height = image.getHeight();

			//Draw image onto panel
			g.drawImage(image, 0, 0, width, height, null);

			//Set panel size
			setPreferredSize(new Dimension(width, height));
		}
	}
}

File: src/main/kotlin/me/jakejmattson/touchcontrol/display/ImagePanel.kt
package me.jakejmattson.touchcontrol.display

import java.awt.*
import java.awt.image.BufferedImage
import javax.swing.JPanel

/**
 * Panel - holds image to display in GUI.
 *
 * @author JakeJMattson
 */
internal class ImagePanel : JPanel() {
    var image: BufferedImage? = null

    public override fun paintComponent(g: Graphics) {
        super.paintComponent(g)

        image ?: return

        val width = image!!.width
        val height = image!!.height

        g.drawImage(image, 0, 0, width, height, null)
        preferredSize = Dimension(width, height)
    }
}